helix-core/src/movement.rs

```diff
@@ -104,6 +104,22 @@ pub fn move_prev_word_start(slice: RopeSlice, range: Range, count: usize) -> Ran
     word_move(slice, range, count, WordMotionTarget::PrevWordStart)
 }
 
+pub fn move_prev_word_end(slice: RopeSlice, range: Range, count: usize) -> Range {
+    word_move(slice, range, count, WordMotionTarget::PrevWordEnd)
+}
+
+pub fn move_this_word_start(slice: RopeSlice, range: Range, count: usize) -> Range {
+    word_move(slice, range, count, WordMotionTarget::ThisWordStart)
```

Comment:
This file is pretty thoroughly unit tested, so it would be good to add tests for these new moves not to lose relative coverage.

Reply:
I added tests for `PrevWordEnd` and removed the `ThisWord` movements since anchors were not positioned correctly for them. Textobjects only need the head anyway so I instead added simple functions to get boundary positions.